### DISCUSSION TITLE
Make ComfyUI friendly for new users. (Desktop v1)

### DISCUSSION THREAD

**Discussion by SunGreen777** [comfyanonymous/ComfyUI#5811] [General]
I am a ForgeUI user, I don’t know that for GGUF models you need to load nodes, I don’t know that they are placed not in checkpoints but in unet. When detected, you can automatically offer to load nodes and move the models to the folder where they belong.
Please tell me how to enable Lora preview in Comfy desktop? I saw this in the presentation version. Also download nodes, which ones?

**Comment by SunGreen777** [comment]
![image](https://github.com/user-attachments/assets/62ee8a5f-f5be-4d79-8dde-bd13db2557d7)
This is very inconvenient, has no one tried to make it convenient in a year? CivitAi Helper example.

**Comment by SunGreen777** [comment]
Comfy has excellent functionality and just a terrible and inconvenient design that turns the creative process into a chore for programmers and not artists.

**Comment by JorgeR81** [comment]
If you're new to ComfyUI, you should install the extension. 
It allows easy installation and management of other extensions. 

https://github.com/ltdrdata/ComfyUI-Manager

![m00](https://github.com/user-attachments/assets/f6b43c3c-ecc6-4cee-af03-ee46f92e8006)


![e1](https://github.com/user-attachments/assets/5d1962e2-510c-4e5a-98a2-4123ecd53eb6)

**Reply by SunGreen777** [reply]
The manager is in the Desktop version of Comfy.
This is a post about how we are constantly being forced to do things we don't have to do. 
There is no problem scanning the catalog and downloading the necessary models for their operation.  There is no problem to inform the user that we have found such and such files, if the work will not be correct, move them to the UNET folder.
No problem to embed an analog of CivitAi helper to make previews for Lora. 
And this is a small part of what I encountered in the first hours.

**Comment by JorgeR81** [comment]
You can use Flux in the Unet version or "checkpoint" version, depending on the model you have downloaded.
The "checkpoint" version already includes Text Encoders and VAE.

In the examples page you can get the all the models and workflows. 
https://comfyanonymous.github.io/ComfyUI_examples/flux/

To share models with other UI's see here:
https://github.com/comfyanonymous/ComfyUI?tab=readme-ov-file#how-do-i-share-models-between-another-ui-and-comfyui

----

To use GGUF models, you need this extension:
https://github.com/city96/ComfyUI-GGUF

You will get a new custom node to load GGUF models ( placed on your Unet model's folder ).
And you also need to use the native ComfyUI nodes, to load the text encoders, and VAE.

![g01](https://github.com/user-attachments/assets/e0a85b52-5bfe-4867-8d00-7895f941c012)

**Comment by JorgeR81** [comment]
To improve the GUI, I recommend these extensions: 

---

**cg-controller**

https://github.com/chrisgoringe/cg-controller

It allows to duplicate nodes, in dockable windows, so you can build a more "traditional" GUI, for each workflow.

See discussion here: https://github.com/comfyanonymous/ComfyUI/discussions/5408


---

**ComfyUI-Custom-Scripts**

https://github.com/pythongosssss/ComfyUI-Custom-Scripts

Among many other features, it has custom nodes to load checkpoints and loras.

- You have a nested combo box, based on the sub folders you create on your models folders.

![p1](https://github.com/user-attachments/assets/7437008a-5806-4621-8bfa-29fb2150044b)


- You can get the Civitai info, for each model or lora.


![l11](https://github.com/user-attachments/assets/14562b0d-6b81-4610-b857-3911949b63d8)

![l1](https://github.com/user-attachments/assets/deb1974c-c587-4d20-bde4-8c666e105ee2)

---

- You can also add an image for each model or lora, that will show up on mouseover.
( that image will be added to your lora models folder )

![pv](https://github.com/user-attachments/assets/8bedee22-805c-48dd-bb4f-3f9616782293)

![d1](https://github.com/user-attachments/assets/c3206a0d-5eb1-4163-bb00-4132d60441d3)


----

**rgthree-comfy**

- https://github.com/rgthree/rgthree-comfy

Among many other features, it will add nested combo boxes to all ComfyUI native loader nodes ( e.g. ControlNet Loader )

![r3](https://github.com/user-attachments/assets/72d2442b-29ef-4e29-b3d6-c7ad7e9febfb)

It also has a custom node, to load multiple loras in a single node.

![r2](https://github.com/user-attachments/assets/a99bfba1-bc48-499a-841d-d563eac34ecc)

This node also allows to get Civitai info, for each lora.

![r3b](https://github.com/user-attachments/assets/fe767c7b-11d7-49c7-b526-31d8771e2ffe)

![r1](https://github.com/user-attachments/assets/ea648e8f-0f3d-4d00-8319-7bdcdc80c0f4)

![r2b](https://github.com/user-attachments/assets/1a2fcf29-37c0-4a3b-923b-cd656b6c0115)

**Comment by JorgeR81** [comment]
To get high-quality-previews of the image, while generating the image, you need to add these models:

https://github.com/comfyanonymous/ComfyUI?tab=readme-ov-file#how-to-show-high-quality-previews

---

You can enable previews directly from the "Manager" extension ( mentioned above )

![m1](https://github.com/user-attachments/assets/76f0e22d-9d2b-4d67-8359-4b198f2ef31f)

**Reply by SunGreen777** [reply]
![Без имени-2](https://github.com/user-attachments/assets/79692817-f1dc-4040-97be-2ced08b9a811)
Thanks for the tips and answers, but it can be done differently. A good example is the browser in Fl Studio 24. See how it is conveniently implemented there.  I have drawn an example.
Tab for Lora tab for nodes and models.
It looks friendly and understandable. You have never used the civitai helper in a1111 civitai helper. 
You can hardly argue that it was inconvenient.

**Reply by LukeG89** [reply]
@SunGreen777 I don't have this node installed, but you could try [ComfyUI-Model-Manager](https://github.com/hayden-fr/ComfyUI-Model-Manager), seems pretty user-friendly

**Reply by JorgeR81** [reply]
If you have ideas on how to improve the GUI, you can make feature requests here:

https://github.com/Comfy-Org/ComfyUI_frontend/issues

----

I don't have the desktop version, but the Sidebar also allows to load models and loras.

![d1](https://github.com/user-attachments/assets/bb6143ba-0387-4233-91d6-12a12f9bdf68)

---

Previews are also possible, but it seems they need to be included in the model ( like in the official SDXL models ).

![d2](https://github.com/user-attachments/assets/ea7f1e38-ac88-43c5-a032-085a338cb0ef)

**Reply by SunGreen777** [reply]
> @SunGreen777 I don't have this node installed, but you could try [ComfyUI-Model-Manager](https://github.com/hayden-fr/ComfyUI-Model-Manager), seems pretty user-friendly
Thanks for the advice, it’s nice to see all these useful options right away during installation, this will make it much easier for new users to use the interface

**Reply by LukeG89** [reply]
> Thanks for the advice, it’s nice to see all these useful options right away during installation, this will make it much easier for new users to use the interface

@SunGreen777 All these options proposed are not part of ComfyUI, they are all custom nodes created by other developers, this is why you can't install them right away. The Manager is a custom node itself, but they made part of the core in the desktop version to make life easier for new users. You can install all these extensions through the Manager.

**Comment by JorgeR81** [comment]
To get the Adetailer in ComfyUI, you also need an extension.
Here it's called FaceDetailer. There is a different node for other areas ( e.g. hands ).

These extensions are from the same developer of the Manager.
The "ComfyUI-Inspire-Pack" has extra features.

https://github.com/ltdrdata/ComfyUI-Impact-Pack
https://github.com/ltdrdata/ComfyUI-Inspire-Pack

![fd](https://github.com/user-attachments/assets/2838957a-9896-48c4-8383-6c0ef131f179)

**Comment by JorgeR81** [comment]
To get IPAdapter for SD1.5 and SDXL, you need these extensions:

https://github.com/cubiq/ComfyUI_IPAdapter_plus
https://github.com/cubiq/ComfyUI_essentials

The "essentials" extension, it's from the same developer and it has many nodes, that make it easier to work with ComfyUI and IPAdapter.

**Comment by SunGreen777** [comment]
The Comfy interface is about experimentation, about comparisons, it doesn't encourage creativity. 
Imagine if Adobe sent you to finish Camera Raw or other tools, would it be popular? 
There are smart people on the Comfy team, but no one who can look through the eyes of a user.
it's all very functional and awkward. I hope the developers will make Comfy Desktop available to most people. 
Thanks again for the tips. I'm going to go hope for a revival of A1111
![image](https://github.com/user-attachments/assets/36078b7f-f791-4b46-b859-8c2d545d48f5)
